primitives/src/communities.rs

```diff
@@ -50,6 +55,48 @@ pub fn validate_nominal_income(nominal_income: &NominalIncome) -> Result<(), ()>
     Ok(())
 }
 
+fn fmt(cid: &CommunityIdentifier, f: &mut Formatter<'_>) -> fmt::Result {
```

Comment:
I doubt this works as expected because the bytes are already base32 encoded as they are geohash bytes. 
Please write a unit test in this file that verifies that
the first 5 characters for a community with location 0 = lat 48.669, lon -4.329 are `gbsuv`
add other examples in all quadrants with the help of https://www.movable-type.co.uk/scripts/geohash.html

Please always first define the struct and add implementations afterwards (just reverse order of your blocks)